program_intent_engagement/apps/core/models.py

```diff
@@ -36,3 +38,44 @@ def get_full_name(self):
 
     def __str__(self):
         return str(self.get_full_name())
+
+   
+class ProgramIntent(TimeStampedModel):
+    """
+    Information about the Program intent
+    
+    .. no_pii:
+    """
+    REASON = (
+        ('CC', 'All Courses Complete'),
```

Comment:
I haven't read about choices recently - is the CC what goes in the DB?

Unlike the standard django choice field this isn't really the result of a dropdown menu, it's an enumeration -maybe django has a better way to do those now?

So I'd prefer to see BIG_CONSTANT_NAMES here like ALL_COURSES_COMPLETE etc
And we don't need a pretty string version, it can just be the same string again.

Reply:
Got it, the first value is what's displayed in the DB and the second part is what's displayed in the form.

We were going off of the example given [here](https://docs.djangoproject.com/en/4.0/topics/db/models/#field-options). I just found another doc now [here](https://docs.djangoproject.com/en/4.0/ref/models/fields/#field-choices) that uses the other format so we can switch it to that with the caps 👍

Reply:
actually I'm going to correct myself - we don't know enough about the set of program intents that will exist to nail it down now. Currently there are three but people will add new ones for sure.

So let's just leave it as a straight string field, non-nullable.